Xamarin.Forms.Core/Page.cs

```diff
@@ -36,8 +36,18 @@ public class Page : VisualElement, ILayout, IPageController, IElementConfigurati
 
 		public static readonly BindableProperty TitleProperty = BindableProperty.Create("Title", typeof(string), typeof(Page), null);
 
+		public static readonly BindableProperty StatusBarColorProperty = BindableProperty.Create(nameof(StatusBarColor), typeof(Color), typeof(Page), Color.Default, 
+			propertyChanged: (bo, o, n) => ((Page)bo).StatusBarColorUpdated());
+
+		public static readonly BindableProperty StatusBarStyleProperty = BindableProperty.Create(nameof(StatusBarStyle), typeof(StatusBarStyle), typeof(Page), StatusBarStyle.Default,
+			propertyChanged: (bo, o, n) => ((Page)bo).StatusBarStyleUpdated());
+
 		public static readonly BindableProperty IconImageSourceProperty = BindableProperty.Create(nameof(IconImageSource), typeof(ImageSource), typeof(Page), default(ImageSource));
 
+		static readonly BindablePropertyKey IsLoadedPropertyKey = BindableProperty.CreateReadOnly(nameof(IsLoaded), typeof(bool), typeof(Page), false);
```

Comment:
Can you make this PR work without adding this property? We're not going to be able to add a property like this as a part of this PR.

We have a spec here https://github.com/xamarin/Xamarin.Forms/issues/8596 for this property

In the code where you are checking for IsLoaded can you instead wire up to the Appearing event or use a local variable? For example in PageRenderer

https://github.com/xamarin/Xamarin.Forms/blob/6fda5941981d4566f54b873d3ab297a76ccac21e/Xamarin.Forms.Platform.Android/Renderers/PageRenderer.cs#L57

You could just set a local property.

Reply:
@PureWeen Is there any news when #8596 will be ready?
Maybe we can change the name?
At first it was IsOnScreen.
Because to call the work of this property, I need to know exactly whether the page is currently on the screen or not.

Reply:
Would it also be a good idea to have the same functionality in 2 implementations?

Reply:
The spec I linked to actually won't tell you what Page is onscreen

From the looks of how you've wired up *IsLoaded* it seems like you're just exposing out _hasAppeared.

I still think doing this a bit more event based around appearing and dispappering would be more resilient though. 

Just wiring up to Appearing inside 
https://github.com/xamarin/Xamarin.Forms/pull/8298/files#diff-5c15636b9e632ed12684a0c5ca1f29e8R165

that way when appearing fires it'll update the statusbarcolors and then locally inside the renderer you can just have a property to track appeared/disappeared


For example in the UAP code the UpdateStatusBar calls that you have are before the renderer sends the appearing event too the cross platform code

https://github.com/xamarin/Xamarin.Forms/pull/8298/files#diff-ebd6a6485423cd0b5a1c9af41db5bea1R68

Reply:
For make this functionality to work correctly, I need a property in the Page class.
So I'm ready to rename it. but we need to come up with a name.
Also, the public property will make it convenient to use it in the ViewModel
@PureWeen

Reply:
> To make this functionality work correctly, I need a property in the Page class.

Why? What behavior in this PR can't be achieved with what I've proposed?

Reply:
The fact is that the color status of the bar is set for the entire application.
And a situation is possible, when:
you create Page1 and Page2.
You put the first page color red, and the second yellow.
and set MainPage = page1;
and the color of the status bar will be yellow. because as soon as you assign the color to page 2, it will be set for the entire application.
And it does not matter whether the page is displayed or we just created an instance of the class.
Another case of MultiPage, such as NaviagationPage or TabbedPage.
There is one page, but inside there are other pages and they switch.

I made a property - which will say whether the page is on the screen now or not.
And set the color only when the page is on the screen.

I think that the property of the page will be the best solution. This does not require invading all possible page renderers.
And adds convenience to the MVVM pattern :)
so, two in one

Also, for the build of this branch, we need the latest Xcode on build agent.

Reply:
> Another case of MultiPage, such as NaviagationPage or TabbedPage.
There is one page, but inside there are other pages and they switch.

I get this and we have this concept already working multiple places
- Shell.SetNavBarIsVisible
- Shell.SetNavBarHasShadow
- NavigationPage.SetHasNavigationBar
- NavigationPage.SetHasBackButton
- ContentPage.ToolBarItems
- etc...

We already have a lot of implementations that effect global features based on the active page in different multi page scenarios so I think we should just tie into those

https://github.com/xamarin/Xamarin.Forms/blob/c579e17ee08665b5dc787b05dd530869e9792b7f/Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs#L119


> IsLoaded

The annoying truth is that adding an IsLoaded property is a very large discussion and if this PR is going to hinge on that property being implemented it will be awhile before this PR can be merged. For example tying a property called IsLoaded to Appearing doesn't really make sense. If you have a TabPage with 3 tabs then technically all 3 of those pages should have IsLoaded set to true even if they aren't visible.

To create the property you want we would need to add a new property like 
*IsPresented* and wire it up differently than the Appearing event.

And even then it's arguable whether you'd want to tie the StatusBarColor change to IsPresented. For example on Shell we tie NavBarVisible to WillShowViewController (https://github.com/xamarin/Xamarin.Forms/blob/c79e187d61501f70ec8cdd40b5d3b447407980b0/Xamarin.Forms.Platform.iOS/Renderers/ShellSectionRenderer.cs#L492)

Otherwise there is a weird delayed animation with the navbar being hidden



> Shell Implementation

In a perfect world the shell implementation would be an attached property that could be applied to Shell,ShellItem, ShellSection, ShellContent, or ContentPage and utilize this mechanism

https://github.com/xamarin/Xamarin.Forms/blob/c79e187d61501f70ec8cdd40b5d3b447407980b0/Xamarin.Forms.Core/Shell/Shell.cs#L1039

Though I could see leaving that for a second PR

Reply:
I renamed the property now it's call IsPresented

Reply:
Relying on the state of the application to be correct during imperative checks of IsPresented is going to be buggy

Tying *IsPresented* to the appearing event is also wrong because Appearing doesn't imply IsPresented it implies that it WILL be Presented 

We already have a lot of cases where the "active page" in multi page scenarios affect global features like the navbar or tabbars

Overall i'm not saying a property like *IsPresented* wouldn't be useful but if we're going to add a property like that it would need to be a separate PR that will probably involve a lot more discussion. 

Also, even if we added *IsPresented* my recommendation would be to add a check here 
https://github.com/xamarin/Xamarin.Forms/pull/8298/files#diff-5c15636b9e632ed12684a0c5ca1f29e8R96

that would basically be

```
else if (e.PropertyName == Page.IsPresented.PropertyName)
				UpdateStatusBarStyle();
```

Even with an *IsPresented* property we'd want the *UpdateStatusBarStyle* sets to happen from the event of IsPresented toggling

Reply:
@PureWeen added some code